### PR TITLE
Feat/sign in flow

### PR DESCRIPTION
**Pull request by GarryFirst-cripto**
Implement Sign In flow on the server

### CODE REVIEW

**Comment by AlikRakhmonov** [comment]
why does it have 73 files? please merge dev into your branch

**Comment by vikachuu** [comment]
@GarryFirst-cripto 
you should probably do the next steps

1. (being on your feature branch)
`git pull origin dev`

2. fix all conflicts if any exists

3. `git add` and `git commit` changes

4. again `git push origin feat/Sign-in-Flow`

The problem only **your** changes should be in PR.

File: backend/src/api/routes/authRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { run } from '../../common/utils/routeHelper';
+import { login, register } from '../../services/authService';
+import authenticationMiddleware from '../middlewares/authenticationMiddleware';
+
+const router = Router();
+
+router
+  .post('/login', authenticationMiddleware, (req, res, next) => login(req.user)
+    .then(data => res.send(data))
+    .catch(next))
```
Comment (vikachuu) [inline backend/src/api/routes/authRoutes.ts:11]: You should use `run` helper method from /common/utils/routeHelper here.

File: backend/src/common/models/user/IRegisterUser.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface IRegisterUser {
+  fullName: string;
+  email: string;
+  password: string;
+}
+
+export interface IUserId {
+  id: string;
+}
```
Comment (vikachuu) [inline backend/src/common/models/user/IRegisterUser.ts:9]: It's better to follow the rule "one file for one interface"

**Pull request closed without merging**